github/src/tools.js

```diff
@@ -143,6 +186,29 @@ export async function listPRComments(octokit, owner, repo, prNumber) {
         repo,
         issue_number: prNumber,
     });
+
+    if (data.length > 10) {
+        try {
+            const gptscriptClient = new GPTScript();
+            const dataset = await gptscriptClient.createDataset(
+                process.env.GPTSCRIPT_WORKSPACE_DIR,
+                `${owner}_${repo}_pr_${prNumber}_comments`,
+                `Comments for PR #${prNumber} in ${owner}/${repo}`
+            );
+            for (const comment of data) {
+                await gptscriptClient.addDatasetElement(
```

Comment:
nit: would be convenient to be able to initialize a dataset with a set of elements and/or add N elements to a dataset in a single statement?

Reply:
I agree that this would be good. It will take some time for me to add that, but that's an improvement I can make while I wait for further feedback on datasets. I'll do this in a follow-up PR.